### PR TITLE
Add development environment using docker compose

### PR DESCRIPTION
**Pull request by yenienserrano**
### Description
Added the development environment to manage issues in Wazuh-dashboard.
 
### Issues Resolved
- #21 

### Test

You must be able to log in to wazuh-dashboards with the environments.

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
  - [x] `yarn test:ftr`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: docker/dev.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+COMPOSE_FILE=dev.yml
+COMPOSE_CMD="docker compose -f $COMPOSE_FILE"
+
+export REPO_PATH="../"
+export SECURITY_PLUGIN_REPO_PATH="../../wazuh-security-dashboards-plugin"
+export WZD_CONFIG_PATH="./config/wzd/opensearch_dashboards.yml"
+export NODE_VERSION=$(cat $REPO_PATH/.nvmrc)
+export OPENSEARCH_VERSION=$1 || $(node -e "console.log(require('$REPO_PATH/package.json').version)")
```
Comment (Desvelao) [inline docker/dev.sh:10]: **thought**: if NodeJS is required to run the script, then maybe we could mention this optional requirement in the readme.

File: docker/dev.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    user: "1000"
+    working_dir: /home/node/wzd
+    networks:
+      - opensearch-net
+    command: >
+      /bin/bash -c "
+        cd /home/node/wzd
+        tail -f /dev/null
+      "
+    # entrypoint: ["tail", "-f", "/dev/null"]
```
Comment (Desvelao) [inline docker/dev.yml:59]: **nitpick**: remove if this is not necessary.

**Review by AlexRuiz7 — CHANGES_REQUESTED** [review]
The proposed solution only works for published versions of OpenSearch as it is based on the Docker images published by them on each release, so we cannot use versions under development using this approach.

I've developed a variant that uses the OpenSearch snapshots, which are generated even for versions under development. The problem is that the snapshots are minimal and do not include any additional plugins, not even the security one, so we cannot use our security plugin in the frontend neither, unlike the solution added here.

As both solutions have advantages and disadvantages, we'll keep both, so we can decide which version to start (with or without security) depending on the situation.

We'll eventually upgrade these tools if we see fit, hopefully solving the limitations of both partial solutions.


```yml
version: "3.9"

services:
  # Runs the bootstrap and exits
  installer:
    image: node:${NODE_VERSION}
    container_name: installer
    volumes:
      - ${REPO_PATH}:/home/node/app
      # - ${SECURITY_PLUGIN_REPO_PATH}:/home/node/app/plugins/security
    user: "1000:1000"
    working_dir: /home/node/app
    command: >
      /bin/bash -c "
        yarn osd bootstrap
      "

  wazuh-dashboard:
    image: node:${NODE_VERSION}
    container_name: wazuh-dashboard
    depends_on:
      installer:
        condition: service_completed_successfully
    ports:
      - 5601:5601 # Map host port 5601 to container port 5601
    expose:
      - "5601" # Expose port 5601 for web access to Wazuh Dashboard
    volumes:
      - ${REPO_PATH}:/home/node/app
      # - ${SECURITY_PLUGIN_REPO_PATH}:/home/node/app/plugins/security
    user: "1000:1000"
    working_dir: /home/node/app
    command: >
      /bin/bash -c "
        yarn opensearch snapshot
      "
```

File: docker/dev.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+if [ "$2" = "security" ]; then
+    export SECURITY_PLUGIN_REPO_PATH="../../wazuh-security-dashboards-plugin"
+    COMPOSE_FILE=docker/dev_security.yml
+
+fi
+
+# Common variables
+export REPO_PATH=$(pwd)
+export NODE_VERSION=$(cat .nvmrc)
+export OPENSEARCH_VERSION=$(bash dev-tools/get-version.sh)
```
Comment (AlexRuiz7) [inline docker/dev.sh:18]: This script is removed in OpenSearch 2.6.0. It's replaced in the `package.json` by Node command instead.

**Review by AlexRuiz7 — CHANGES_REQUESTED** [review]

File: docker/dev.sh (hunk 3 above)
Comment (AlexRuiz7) [reply docker/dev.sh:18]: As we want the environment to be node-agnostic, using the Node command would not be a solution for use.

Instead, I'll port the bash command in `dev-tools/get-version.sh`. The method to read the OpenSearch version in the `package.json` file will diverge from OpenSearch's, but we are fine with it.

Comment (AlexRuiz7) [reply docker/dev.sh]: Solved in https://github.com/wazuh/wazuh-dashboard/pull/30/commits/ef942756ac5eacecc2fabc25d71bbd6bfa7551a6

**Review by Desvelao — APPROVED** [review]
**review**

:green_circle: code

**Pull request merged**